_posts/2019-12-25-technical-debt.md

```diff
@@ -0,0 +1,51 @@
+
+> i have a theory, which is that we struggle to get the time allocated to pay down technical debt (or improve deploys, etc) because to biz types we basically sound like the underpants gnomes. step 1 ... pay down technical debt. step 2 ... step 3 .. PROFIT - Charity Majors, CTO @ Honeycomb.io
+
+Pursuading stakeholders to pay down technical debt is fraught with difficulty. Requesting to pay down technical debt is akin to asking them to trade away something they can confidently say has some value, for a loose promise that if and when they ask for something else, it will be easier to deliver. It is little wonder that most technical debt doomsaying fails to convince anyone, notably in the low-trust terrain that forms the modern enterprise. The benefits of paying down technical debt are often presented in a nebulous way, which is in contrast to the concrete nature of getting a feature out to market sooner. Technical debt may be a murky concept, but that does not mean that the benefits of paying it down are. If we want technical debt to be taken in earnest, we need to get better at gauging the effects and presenting a case for removal that has discernable business benefit.
+
+It is impossible to measure the intangible technical debt, but this property does not always extend to its effects. We can find another measurement (or more) that can act as a suitable proxy. This is not too dissimilar to maintenance practices in heavy industry. You can never know for certain when a piece of equipment is going to fail, but you can measure how well maintenance crews are adhering to a planned maintenance schedule. In most cases this is a reasonable indication of equipment health*. What measures could we identify that could act as reasonable proxies for the amount of technical debt across one or more software systems?
+
+I believe a suitable proxy for technical debt can be found in existing literature on software delivery. The State of DevOps Report produced by DevOps Research and Assessment (hereby refered to as "the DORA Report") is compiled annually and details the habits of high performing software teams. It is one of the most rigourous studies on software development, produced from surveying and observing hundreds of teams across the world. If we are to come upon a satisfactory measure of technical debt, it would likely be in the DORA report. The report distinguishes two key measures of software delivery performance; throughput and stability. Throughput is measured via deployment frequency and lead time for changes. Stability is measured via time to restore a service and the change failure rate. I believe that change lead time and time to restore service are useful signals of the level of technical debt a project has acrued. Both of these marry well with the effects of technical debt discussed earlier, which included decreased velocity and increasingly risky changes.* Both of these measures are also difficult to game.*
```

Comment:
difficult to game = new para with more words. there's something you're not saying here?

Reply:
I was deciding whether or not move that topic to the following paragraphs. In the end I decided to write another paragraph detailing what a good proxy measurement should have, and inserted that there.